res/controllers/Numark-NS6II-scripts.js

```diff
@@ -0,0 +1,1479 @@
+        },
+        shift: function() {
+            this.inKey = "rateRange";
+            this.input = NS6II.makeButtonDownInputHandler(function() {
+                this.inSetValue(rates.next());
+            });
+        },
+        outConnect: false,
+    });
+    this.pitchBendMinus = new components.Button({
+        midi: [0x90 + channelOffset, 0x0C],
+        // shift: [0x90+channelOffset,0x2C]
+        unshift: function() {
+            this.inKey = "rate_temp_down";
+            this.input = components.Button.prototype.input;
+        },
+        shift: function() {
+            this.inKey = "rateRange";
+            this.input = NS6II.makeButtonDownInputHandler(function() {
+                this.inSetValue(rates.previous());
+            });
+        },
+        outConnect: false,
+    });
+    this.shiftButton = new components.Button({
+        midi: [0x90 + channelOffset, 0x20],
+        input: function(channelmidi, control, value, status, _group) {
+            if (this.isPress(channelmidi, control, value, status)) {
+                NS6II.mixer.shift();
+                NS6II.EffectUnits[channelOffset % 2 + 1].shift();
+                theDeck.shift();
+            } else {
+                NS6II.mixer.unshift();
+                NS6II.EffectUnits[channelOffset % 2 + 1].unshift();
+                theDeck.unshift();
+            }
+        },
+    });
+
+    this.sync = new components.SyncButton({
+        midi: [0x90 + channelOffset, 0x02],
+        // shift: [0x90+channelOffset,0x03]
+    });
+
+    this.play = new components.PlayButton({
+        midi: [0x90 + channelOffset, 0x00],
+        // shift: [0x90+channelOffset,0x04]
+    });
+    this.cue = new components.CueButton({
+        midi: [0x90 + channelOffset, 0x01],
+        // shift: [0x90+channelOffset,0x05]
+    });
+
+    // midi: [0xB0 + channelOffset, 0x06],
+    this.jog = new components.JogWheelBasic({
+        deck: deckNumber,
+        wheelResolution: 1000, // measurement (1000) wasn't producing accurate results (alt: 1073)
+        alpha: NS6II.SCRATCH_SETTINGS.alpha,
+        beta: NS6II.SCRATCH_SETTINGS.beta,
+    });
+
+    this.stripSearch = new components.Pot({
+        midi: [0xB0 + channelOffset, 0x4D], // no feedback
+        // input MSB: [0xB0+deck,0x2F] LSB
+        group: theDeck.group,
+        inKey: "playposition",
+        shift: function() {
+            this.inSetParameter = components.Pot.prototype.inSetParameter;
+        },
+        unshift: function() {
+            this.inSetParameter = function(value) {
+                // only allow searching when deck is not playing.
+                if (!engine.getParameter(this.group, "play")) {
+                    engine.setParameter(this.group, this.inKey, value);
+                }
+            };
+        },
+    });
+    this.scratch = new components.Button({
+        midi: [0x90 + channelOffset, 0x07],
+        // shift: [0x90+channelOffset,0x46]
```

Comment:
Can this be removed as this is redundant with line 409/414?

Reply:
technically yes, but I would've preferred this to stay here instead of having it buried in the implementation.